repo-info/get_repo_info.py

```diff
@@ -0,0 +1,62 @@
+import os
+import argparse
+import datetime
+import pandas as pd
+from tqdm import tqdm
+from github import Github
+
+tqdm.pandas()
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-t', '--github_access_token', help='GitHub access token to overcome API rate limitations')
+parser.add_argument('-f', '--filepath', help='Filepath of .csv file containing repo data')
+parser.add_argument('-c', '--colname', help='Column name in CSV file pertaining to repo URL')
+args = parser.parse_args()
+
+GITHUB_API_RATE_LIMIT = 5000
+FILEPATH, COLNAME, GITHUB_ACCESS_TOKEN = args.filepath, args.colname, args.github_access_token
+
+data = pd.read_csv(FILEPATH)
+data = data[data['Status'].isna()]
+REPO_URLS = data[COLNAME].unique()
+NUM_REPOS = REPO_URLS.shape[0]
+
+def check_number_repos():
+    if NUM_REPOS > GITHUB_API_RATE_LIMIT:
+        print(f'You can only make {GITHUB_API_RATE_LIMIT} requests per hour. Your file has {NUM_REPOS} unique repositories. Exiting.')
+        exit(0)
+
+def get_diff_month(d1, d2):
+    return (d1.year - d2.year) * 12 + d1.month - d2.month
+
+def get_repo_object(repo_url):
+    try:
+        repo_name = repo_url.split('github.com/')[1]
+        return Github(GITHUB_ACCESS_TOKEN).get_repo(repo_name)
+    except Exception as e:
+        print(e)
+        return None
+
+def get_months_since_last_commit(repo):
+    try:
+        master_branch = repo.get_branch('master')
```

Comment:
Are you trying to find the default branch name? Please see https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-branches#about-the-default-branch

Reply:
Are you saying the default branch to get should be `'main'` instead of `'master'`?

Reply:
Different repos call the default branch differently. You can ask for the "default branch" instead of a specific name.